src/NuGet.Core/NuGet.Protocol/Resources/RawSearchResourceV3.cs

```diff
@@ -113,6 +84,68 @@ public virtual async Task<JObject> SearchPage(string searchTerm, SearchFilter fi
             throw new FatalProtocolException(Strings.Protocol_MissingSearchService);
         }
 
+        private static StringBuilder LastStringBuilder;
+
+        private string GenerateQueryString(string searchTerm, SearchFilter filters, int skip, int take)
+        {
+            // Use a cached string builder, if available.
+            var queryStringBuilder = Interlocked.Exchange(ref LastStringBuilder, null);
+            if (queryStringBuilder == null)
+            {
+                queryStringBuilder = new StringBuilder(128);
+            }
+            else
+            {
+                queryStringBuilder.Clear();
+            }
+
+            queryStringBuilder.Append("q=");
+            queryStringBuilder.Append(searchTerm);
+            queryStringBuilder.Append("&skip=");
+            queryStringBuilder.Append(skip.ToString());
+            queryStringBuilder.Append("&take=");
+            queryStringBuilder.Append(take.ToString());
+            queryStringBuilder.Append("&prerelease=");
+            queryStringBuilder.Append(filters.IncludePrerelease ? "true" : "false");
+
+            if (filters.IncludeDelisted)
+            {
+                queryStringBuilder.Append("&includeDelisted=true");
+            }
+
+            if (filters.SupportedFrameworks != null
+                && filters.SupportedFrameworks.Any())
+            {
+                foreach (var framework in filters.SupportedFrameworks)
+                {
+                    queryStringBuilder.Append("&supportedFramework=");
+                    queryStringBuilder.Append(framework);
+                }
+            }
+
+            if (filters.PackageTypes != null
+                && filters.PackageTypes.Any())
+            {
+                foreach (var type in filters.PackageTypes)
+                {
+                    queryStringBuilder.Append("&packageTypeFilter=");
+                    queryStringBuilder.Append(type);
+                }
+            }
+
+            queryStringBuilder.Append("&semVerLevel=2.0.0");
+
+            var queryString = queryStringBuilder.ToString();
+
+            // Only cache the string builder if it doesn't use too much memory
+            if (queryStringBuilder.Capacity <= 1024)
+            {
+                Interlocked.Exchange(ref LastStringBuilder, queryStringBuilder);
```

Comment:
`Clear()` first?

Reply:
[`Clear()`](https://github.com/dotnet/corefx/blob/a533a0fad2c101881945824570dc21845783366a/src/Common/src/CoreLib/System/Text/StringBuilder.cs#L432-L436) is nothing more than an alias of `.Length = 0`, and [`.Length`'s setter](https://github.com/dotnet/corefx/blob/a533a0fad2c101881945824570dc21845783366a/src/Common/src/CoreLib/System/Text/StringBuilder.cs#L447-L504) does not reduce capacity.

So, if the goal is to avoid caching large amounts of memory, calling `.Clear()` won't help. If the goal is to ensure that the next time the `StringBuilder` is used it doesn't append on the end of the previous usage, I prefer to call `Clear()` on acquisition.